nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/FetchElasticsearchHttp.java

```diff
@@ -0,0 +1,293 @@
+        relationships.add(REL_NOT_FOUND);
+        return Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ES_URL);
+        descriptors.add(PROP_SSL_CONTEXT_SERVICE);
+        descriptors.add(CONNECT_TIMEOUT);
+        descriptors.add(RESPONSE_TIMEOUT);
+        descriptors.add(DOC_ID);
+        descriptors.add(INDEX);
+        descriptors.add(TYPE);
+        descriptors.add(FIELDS);
+
+        return Collections.unmodifiableList(descriptors);
+    }
+
+
+    @OnScheduled
+    public void setup(ProcessContext context) {
+        super.setup(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = null;
+        if (context.hasIncomingConnection()) {
+            flowFile = session.get();
+
+            // If we have no FlowFile, and all incoming connections are self-loops then we can continue on.
+            // However, if we have no FlowFile and we have connections coming from other Processors, then
+            // we know that we should run only if we have a FlowFile.
+            if (flowFile == null && context.hasNonLoopConnection()) {
+                return;
+            }
+        }
+
+        OkHttpClient okHttpClient = getClient();
+
+        if (flowFile == null) {
+            flowFile = session.create();
+        }
+
+        final String index = context.getProperty(INDEX).evaluateAttributeExpressions(flowFile).getValue();
+        final String docId = context.getProperty(DOC_ID).evaluateAttributeExpressions(flowFile).getValue();
+        final String docType = context.getProperty(TYPE).evaluateAttributeExpressions(flowFile).getValue();
+        final String fields = context.getProperty(FIELDS).isSet()
+                ? context.getProperty(FIELDS).evaluateAttributeExpressions(flowFile).getValue()
+                : null;
+
+        final ComponentLog logger = getLogger();
+
+
+        try {
+            logger.debug("Fetching {}/{}/{} from Elasticsearch", new Object[]{index, docType, docId});
+
+            // read the url property from the context
+            final String urlstr = StringUtils.trimToEmpty(context.getProperty(ES_URL).getValue());
+            final URL url = buildRequestURL(urlstr, docId, index, docType, fields);
+            final long startNanos = System.nanoTime();
+
+            final Response getResponse = sendRequestToElasticsearch(okHttpClient, url, "GET", null);
+            final int statusCode = getResponse.code();
+
+            if (isSuccess(statusCode)) {
+                ResponseBody body = getResponse.body();
+                final byte[] bodyBytes = body.bytes();
+                JsonNode responseJson = parseJsonResponse(new ByteArrayInputStream(bodyBytes));
```

Comment:
The response contains the index, type, id, and found (and the source if it is found). I could add these as attributes (in case EL is used for these properties) but provenance would already capture this.